client/src/components/history/history-item.tsx

```diff
@@ -73,15 +76,67 @@ export function ChatItem(props: {
   }
 
   function LinkRenderer(props: { href: string; children: React.ReactNode }) {
-    return (
-      <h3>
-        <a href={props.href} target="_blank" rel="noreferrer">
-          {props.children}
-        </a>
-      </h3>
+    const linkAnswer =
+      props.href.length > 30 ? props.href.slice(0, 30) : props.href;
+    return message.text.includes(prefix) ? (
+      <a
+        href="#"
+        rel="noreferrer"
+        onClick={sentQuestion}
+        data-cy={`question-link-${i}`}
+      >
+        {props.children}
+      </a>
+    ) : (
+      <a
+        href={props.href}
+        target="_blank"
+        rel="noreferrer"
+        onClick={sentQuestion}
+      >
+        {linkAnswer}
+      </a>
     );
   }
 
+  function sentQuestion() {
+    const answerArray = message.text.match(/[^()]+/g);
```

Comment:
this isn't correct or narrow enough. It will match any parenthesis in the answer text. You need a proper capture group, e.g.

```typescript
const REGEX_ASK_LINK = new Regex("\(ask:\/\/([^)]+)")


function getFirstAskLinkContent(text: string): string {
  const matches = REGEX_ASK_LINK.match(text)
  return (matches && matches.length > 1) 
    ? matches[1]  // this might be an object and you need one of its props, check JS docs
    : ""  // empty string means no ask links
}
```